### ISSUE TITLE
[Proposal] Add new API to pass options to complete command

### ISSUE THREAD

**Issue by vadmeste** [posener/complete#101]
Hey.

Currently, using posener/complete does not automatically escape spaces in bash.

I think the best solution of this is to add a new API which pass options to complete command, such as:
- `-o filenames` to automatically escapes spaces
- `-o nospace` related to this issue https://github.com/posener/complete/issues/87

What do you think @posener ?

**Comment by posener** [comment]
Hey, I'm currently on vacation. I'll get back to you in a couple of weeks.
I think it should be added to the install stage. If I understand correctly.
Cheers, Eyal

**Comment by posener** [comment]
Hey

If one would want to add these options, they would be added to the bash complete command here: https://github.com/posener/complete/blob/master/cmd/install/bash.go#L36.
* I am not sure if you can add them to other supported shells (fish, zsh).
* You would have to add them also to the install API here: https://github.com/posener/complete/blob/master/cmd/install/install.go#L21
* Then, to the CLI struct: https://github.com/posener/complete/blob/master/cmd/cmd.go#L15, and propagate to the instsall call: https://github.com/posener/complete/blob/master/cmd/cmd.go#L43.

Would you be interested in contributing this options?

**Comment by vadmeste** [comment]
> Would you be interested in contributing this options?

Yes, I am interested.

> * I am not sure if you can add them to other supported shells (fish, zsh).

I think the best thing to do is to add an API which adds more flags and arguments to the completion command. fish complete command is different to bash & zsh

**Comment by AdamSLevy** [comment]
I looked into this some time ago and found that it is not possible to call bash's builtins from an external program. So unless the fundamental mechanism that this completion library operates on is changed, this will not be possible IIRC.

See https://stackoverflow.com/questions/55842705/is-it-possible-to-modify-compopts-dynamically-from-an-external-completion-comman

**Comment by vadmeste** [comment]
> I looked into this some time ago and found that it is not possible to call bash's builtins from an external program. So unless the fundamental mechanism that this completion library operates on is changed, this will not be possible IIRC.
> 
> See https://stackoverflow.com/questions/55842705/is-it-possible-to-modify-compopts-dynamically-from-an-external-completion-comman

Indeed. This is not a dynamic thing. It will be configured during the setup of the completion
